dotCMS/src/main/java/com/dotcms/browser/BrowserQuery.java

```diff
@@ -0,0 +1,225 @@
+        private String filter = null;
+        private List<String> mimeTypes=new ArrayList<>();
+        private List<String>  extensions=new ArrayList<>();
+        private String sortBy="moddate";
+        private int offset=0;
+        private int maxResults=100;
+        private boolean showWorking=true;
+        private boolean showArchived=false;
+        private boolean showFolders=false;
+        private boolean showFiles=false;
+        private boolean sortByDesc=false;
+        private boolean showLinks=false;
+        private boolean showPages=false;
+        private boolean showDotAssets=false;
+        private long languageId=0;
+
+        private Builder() {}
+
+        private Builder(BrowserQuery browserQuery) {
+            this.user = browserQuery.user;
+            this.hostFolderId = browserQuery.hostFolderId;
+            this.filter = browserQuery.filter;
+            this.mimeTypes = browserQuery.mimeTypes;
+            this.extensions = browserQuery.extensions;
+            this.sortBy = browserQuery.sortBy;
+            this.offset = browserQuery.offset;
+            this.maxResults = browserQuery.maxResults;
+            this.showWorking = browserQuery.showWorking;
+            this.showArchived = browserQuery.showArchived;
+            this.showFolders = browserQuery.showFolders;
+            this.showFiles = browserQuery.showFiles;
+            this.showPages = browserQuery.showPages;
+            this.sortByDesc = browserQuery.sortByDesc;
+            this.showLinks = browserQuery.showLinks;
+            this.showDotAssets = browserQuery.showDotAssets;
+            this.languageId = browserQuery.languageId;
+        }
+
+        public Builder withUser(@Nonnull User user) {
+            this.user = user;
+            return this;
+        }
+        
+        public Builder inHostOrFolder(@Nonnull Folder folder) {
+            this.hostFolderId = folder !=null ? folder.getInode() : null;
+            return this;
+        }
+        public Builder inHostOrFolder(@Nonnull Host host) {
+            this.hostFolderId = host.getIdentifier();
+            return this;
+        }
+        public Builder withHostId(@Nonnull String hostid) {
+            this.hostFolderId = hostid;
+            return this;
+        }
+        public Builder withHostOrFolderId(@Nonnull String hostFolderId) {
+            this.hostFolderId = hostFolderId;
+            return this;
+        }
+
+        public Builder withFilter(@Nonnull String filter) {
+            this.filter = filter;
+            return this;
+        }
+
+        public Builder showMimeTypes(@Nonnull List<String> mimeTypes) {
+            this.mimeTypes = mimeTypes;
+            return this;
+        }
+
+        public Builder showExtensions(@Nonnull List<String>  extensions) {
+            this.extensions = extensions;
+            return this;
+        }
+
+        public Builder sortBy(@Nonnull String sortBy) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'sortBy' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337824)